pkg/infrastructure/azure/azure.go

```diff
@@ -122,20 +122,27 @@ func (p *Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionI
 	if err != nil {
 		return fmt.Errorf("failed to get azure resource groups factory: %w", err)
 	}
+	// Check if the resource group exists. Create or update function replaces metadata
+	// like existing tags if the resource group exists which is not what we want so
+	// adding a check to see if it exists and do nothing if it does.
```

Comment:
I added 90f61a065e0dac2ba0f9ea54329e688a1fd643cc to  https://github.com/openshift/installer/pull/9538 which drops this code and uses CAPZ to reconcile the resource group, so I think we can drop this commit in favor of that.

Reply:
@jinyunma FYI I'll be dropping the fix in favour of Patrick's PR.